File: node/src/Classifier1.js
import NNet from "../lib/NNet.js";

function test(){
    //console.time("total")
    const net = new NNet(5)
    net.addLayer(10)
    net.addLayer(2)
    net.batchSize = 10
    net.step = 10
    net.feed([1,0,0,2,1])
    console.log(net.getOutput())
    console.time()

    net.train([1,0,0,2,1],[0,1])
    console.log(net.getOutput())
    net.train([0, 1, 0, 0, 0], [1, 0])
    console.log(net.getOutput())
    //console.log(net)
    console.time("train")

    for (let i=1 ; i<100000;i++) {

        net.train([1, 0, 0, 2, 1], [0, 1])
        net.train([1, 0, 0, 2, 0], [0, 1])
        net.train([0, 1, 0, 0, 1], [1, 0])
        net.train([0, 1, 0, 0, 0], [1, 0])
        net.train([0, 1, 0, 0, 0], [1, 0])
    }
    console.timeLog("train")
    console.log(net)
    net.feed([1,0,0,2,1])
    console.log("expected  [0 , 1] ",net.getOutput())
    net.feed([0,1,0,0,0])
    console.log("expected  [1 , 0] ",net.getOutput())
    net.feed([0,0,0,0,0])
    console.log("expected  [????] ",net.getOutput())

    console.log("Network "+net.layers+" layers "+ net.neurons.length+" neurons  ("+net.allNeurons.length+") "+net.allNeurons.reduce((prev, n)=>(prev+n.in.length),0)+" weights" )

    console.log("Avg Cost "+ net.getAverageCost().toFixed(5))

    console.time("test")
    for (let i=0;i<1000000;i++){
        net.feed([0,1,0,0,0]);
        net.getOutput();
    }
    console.timeLog("test")


    console.timeEnd()
}

test()

File: java/src/Classifier1.java
import java.util.Arrays;

public class Classifier1 {

    //console.time("total")
    private NNet net;

    public Classifier1() {
        net = new NNet(5);
        net.addLayer(50);
        net.addLayer(2);
        net.batchSize = 10;
        net.step = 2;
    }

    public void train(int iter) {
        /*net.feed(new double[]{1.0,0.0,0.0,2.0,1.0});
        System.out.println(net.getOutput());*/
        long startTime = System.currentTimeMillis();
        for (int i = 1; i < iter; i++) {
            net.train(new double[]{1, 0, 0, 2, 1}, new double[]{0, 1});
            net.train(new double[]{1, 0, 0, 2, 0}, new double[]{0, 1});
            net.train(new double[]{0, 1, 0, 0, 1}, new double[]{1, 0});
            net.train(new double[]{0, 1, 0, 0, 0}, new double[]{1, 0});
            net.train(new double[]{1, 0, 0, 2, 1}, new double[]{0, 1});
        }
        System.out.println("train " + (System.currentTimeMillis() - startTime));
    }

    public void test(double[] values, String expected) {
        net.feed(values);
        System.out.println("expected   "+expected+"  "+ Arrays.toString(this.net.getOutput()));
    }

    public static void main(String[] args) {
        Classifier1 c = new Classifier1();
        c.train(100000);
        c.test(new double[]{1,0,0,2,1},"[0,1]");
        c.test(new double[]{1,0,0,2,0},"[0,1]");
        c.test(new double[]{0,1,0,0,0},"[1,0]");
        c.test(new double[]{0,0,0,0,0},"[???]");
        long startTime = System.currentTimeMillis();
        for (int i=0;i<1000000;i++){
            c.net.feed(new double[]{0,1,0,0,0});
            c.net.getOutput();
        }
        System.out.println("Checks "+(System.currentTimeMillis()-startTime));
        System.out.println(c.net.trainings+"   "+c.net.count);
    }
}
